### DISCUSSION TITLE
WhatsApp has improved the ban rules for the message automation system

### DISCUSSION THREAD

**Discussion by jeffersonsc** [tulir/whatsmeow#567] [WhatsApp protocol Q&A] [answered]
1. Does anyone have any reports of recent WhatsApp bans using whatsmeow?
2. Do we have any way to mitigate this ban?

**Comment by BastianOrth2** [comment, marked as answer]
1. yes
2. you can write the support if your behavior was not too obvious and maybe you'll get unbanned. Sometimes it can take weeks to get unbanned but's not recommended to use whatsmeow right after. Maybe after couple of weeks you can try again, but as you're flagged the risks of getting a final ban is quite high. 

Make sure you message output / minute is not too high and make sure that people write you first before you start to write them. But got recently 2 disconnects stating that those numbers use a unofficial WA Version, so yeah... All the precautions not working so well

**Reply by jeffersonsc** [reply]
Perfect, thanks for the responses. I always advise people to warm up the numbers using the app and WhatsApp Web. I try to maintain client names as Chrome, the most popular browser.

In terms of sending, I always try to space messages between 1 and 5 seconds, varying a random value within this range. Whenever I send text, I use the typing indicator, and for OGG audio, I also use the recording indicator, simulating human behavior well. I always advise people to open a connection; in my case, it's just chatbots, so I have a smoother operation.

I see that there are some new notifications when I make a new critical block connection, but later I realized it was just warnings that don't impact much. But anything like that worries us.

**Reply by BastianOrth2** [reply]
Send you a linkedin requests, lets exchange some experiences if you like.

**Reply by williank** [reply]
How to use the typing simulator? @jeffersonsc

**Reply by jeffersonsc** [reply]
I'm using an randon interval between 1000 and 5000, and send typing after send text message or audio message.

**Reply by glpzzz** [reply]
I'm about to start using this library to send what I think is going to be very few messages (1 per week) but to around 300 different numbers. 

I have read the suggestions on this thread. Something else to consider here?

**Comment by hrizal** [comment]
At first I was curious, then I tried just connecting the new number to the
official WhatsApp web using Google Chrome, and started by sending messages.
When I sent approximately 5 messages to the number, my WhatsApp was banned.
I came to the conclusion that this ban rule is not related to whatsmeow,
but applies to the 'possibility' of being considered a spammer

Pada Jum, 12 Apr 2024 pukul 14.04 BastianOrth2 ***@***.***>
menulis:

**Reply by BastianOrth2** [reply]
You should always warm up a number before connecting to wahtsmeow. Best is that some of your friends send you some messages to your number.

**Reply by jeffersonsc** [reply]
True, some clients who had banned numbers and then recovered and connected them received the message that they were not using an official app.
